Inline Support Links: update legacy links to context format

michaeldcain opened the pull request:
In #54991 we switched the `InlineSupportLink` component to use an object of support links and a `supportContext` prop, and in #56661 we added a Tracks event that records that context. Since the legacy events all pass a `null` context, the event can be hard to segment, so this PR switches most of the legacy support links to use the new `supportContext` prop.

**To test:**
- run `localStorage.setItem('debug', 'calypso:analytics*');` on your browser console and filter for "recording event"
- visit the updated links (e.g. Importers, Podcasting, or Customer Home cards)
- click on an inline support link
- verify that the correct support document is shown in a modal
- verify that the `calypso_inlinesupportlink_click` event is fired and the `support_context` prop has a non-null value

michaeldcain committed:
Update themes upload support link

michaeldcain committed:
Update theme switch support link

michaeldcain committed:
Update unsupported theme support link

michaeldcain committed:
Update webmaster tools support link

michaeldcain committed:
Update podcasting support link

michaeldcain committed:
Update portfolio and testimonial support links

michaeldcain committed:
Update google analytics measurement id support link

michaeldcain committed:
Update google analytics measurement id support link

michaeldcain committed:
Update paid newsletters support link

michaeldcain committed:
Update menus support link

michaeldcain committed:
Update stats support links

michaeldcain committed:
Update getting started video support links

michaeldcain committed:
Update importers support links

Aurorum commented on client/components/inline-support-link/context-links.js:
```diff
@@ -83,4 +141,8 @@ export const getContextLinks = () => ( {
 		post_id: 76237,
 	},
+	'webmaster-tools': {
+		link: 'https://wordpress.com/support/webmaster-tools/',
```
Would it be possible to change this to https://wordpress.com/support/site-verification-services instead? It would handle part of #56560 which has already been approved, so hopefully shouldn't be an issue. 

```suggestion
		link: 'https://wordpress.com/support/site-verification-services/',
```

I think that `webmaster-tools` should also become `site-verification-services`, but that might not matter until #56560 is merged anyway.

sixhours replied:
I'll merge #56560 once tests pass so this PR can be rebased against trunk. :)

michaeldcain committed:
Alphabetize

sirbrillig committed:
Change context-links to export a static dictionary

Exporting a function that returned a dictionary means that the
dictionary will be re-created every time the function is called. Since
the dictionary doesn't contain any dynamic data, this is unnecessary and
we can just export the dictionary itself.

sirbrillig committed:
Fix linting issue in README

sirbrillig committed:
Lazy load link and post_id from context

sirbrillig committed:
Fix dynamic import of default export

sirbrillig reviewed (approved):
I wasn't able to find the actual links mentioned here but I verified that the post_ids and links seem to all be correct, and that the inline links for other pages appear and work.
